src/types/text_imput_v3.zig

```diff
@@ -0,0 +1,113 @@
+const wlr = @import("../wlroots.zig");
+
+const wayland = @import("wayland");
+const wl = wayland.server.wl;
+
+pub const TextInputV3Manager = extern struct {
+    global: *wl.Global,
+    text_inputs: wl.list.Head(TextInputV3, "link"),
+
+    server_destroy: wl.Listener(*wl.Server),
+
+    events: extern struct {
+        text_input: wl.Signal(*wlr.TextInputV3),
+        destroy: wl.Signal(*wlr.TextInputV3Manager),
+    },
+
+    extern fn wlr_text_input_v3_create(server: *wl.Server) ?*wlr.TextInputV3Manager;
+    pub fn create(server: *wl.Server) !*wlr.TextInputV3Manager {
+        return wlr_text_input_v3_create(server) orelse error.OutOfMemory;
+    }
+};
+
+pub const TextInputV3 = extern struct {
+    pub const Features = extern struct {
```

Comment:
This struct has no fields, therefore it is a namespace and snake_case. The extern is also unnecessary.
```suggestion
    pub const features = struct {
```